front/src/applications/operationalStudies/components/SimulationResults/DriverTrainSchedule/DriverTrainScheduleHelpers.tsx

```diff
@@ -1,29 +1,32 @@
 import React from 'react';
 import * as d3 from 'd3';
 import nextId from 'react-id-generator';
+import { PositionSpeedTime, SpeedPosition, Stop, Train } from 'reducers/osrdsimulation/types';
 
-export function massWithOneDecimal(number) {
+export function massWithOneDecimal(number: number) {
   return Math.round(number / 100) / 10;
 }
 
-function getTime(sec) {
+function getTime(sec: number) {
   const timeplus = new Date(sec * 1000);
   const time = timeplus.toISOString().substr(11, 8);
-  if (time[6] >= 0 && time[6] < 2) {
-    if (time[6] === '1') {
-      if (time[7] <= 4) {
+  const sixthDigit = parseInt(time[6], 10);
+  const seventhDigit = parseInt(time[7], 10);
+  if (sixthDigit >= 0 && sixthDigit < 2) {
+    if (sixthDigit === 1) {
+      if (seventhDigit <= 4) {
         return time.slice(0, 5);
       }
       return `${time.slice(0, 5)}+`;
     }
     return time.slice(0, 5);
   }
-  if (time[6] >= 1 && time[6] < 5) {
-    if (time[6] === '4') {
-      if (time[7] <= 4) {
+  if (sixthDigit >= 1 && sixthDigit < 5) {
```

Comment:
There is a typo here, I think it should be `sixthDigit > 1` because "=1" is already checked at line 16.

But I had a hard time understanding the whole getTime function (Theo explained me that it should add a "+" at the end of the schedule if the second is between 15 and 44 and the minute should increment if second >= 45).
Do you think it's ok if we change the function to be like that : 
```suggestion
function getTime(sec) {
  let timeplus = new Date(sec * 1000);
  let time = timeplus.toISOString().substr(11, 8);
  const seconds = Number(time.slice(-2));
  if (seconds >= 15 && seconds <= 44) {
    return `${time.slice(0, 5)}+`;
  }
  if (seconds >= 45) {
    const minInMs = 60 * 1000;
    timeplus = new Date(sec * 1000 + minInMs);
    time = timeplus.toISOString().substr(11, 8);
  }
  return `${time.slice(0, 5)}`;
}
``` 
? :) (seems to be still working as expected for me and more understandable)

Reply:
Make another PR to isolate that specifically, This really needs to be tested before any such refactoring 
@SharglutDev

Reply:
Ok, I won't touch this !